### ISSUE TITLE
Using dtwclust with RODBC (database connection)

### ISSUE THREAD

**Issue by xtianfall** [asardaes/dtwclust#13]
Hi, I'm trying to use RODBC to connect to my database and work with a very big data (>15.000.000 rows and 60 columns) and as I expected, the memory of my computer is not enough for download all the data.

I don't know if is possible to use that package with this, or if there is someone that can replace the database connection.

I'll appreciate some help here ;)

**Comment by asardaes** [comment]
Sadly, I doubt I could help you here. R itself needs the data to fit in RAM, and I don't know if there are easy-to-use approaches to apply existing algorithms/packages in an out-of-core fashion.

**Comment by xtianfall** [comment]
I'm trying to use ffdf package to store in the disk. I hope it works like i expect. I'll keep you informed ;)

**Comment by asardaes** [comment]
That could work if it's transparent enough, I suppose.

Even then, 15 million series is quite a lot. It might be easier to create clusters based on a subset and then use the resulting centroids to "classify" the rest of the data. Or maybe a [more elaborate approach](http://www.francois-petitjean.com/Research/Petitjean2017-SDM.pdf) might work for you (even if you're not using DTW)?

**Comment by xtianfall** [comment]
I've tried using ffdf package with yours and it works perfectly ^^ (I just try with small subset, but I think that there is no problem with bigger set).

Yes, 15M is a very big set and I'll use a half do something like this, I've to do more test and have a meeting for use the supercomputer for the final calculations.

Furthermore, I've been cleaning the database and I've found some errors and I'll try a different approach, basically I've a lot of missing values and I don't know how to deal with them without loosing the curve... I've proven, using different length, but I don't get what I speak...

I'll read the PDF and I'll see what I do. Thanks ;)
